In [24]:
from collections import defaultdict
import numpy as np
from mlxtend.preprocessing import TransactionEncoder
from mlxtend.frequent_patterns import apriori
import pandas as pd


dfGender = pd.read_csv('./datasetlab2/BlackFriday.csv', usecols=['Age'])
dfGender['Age'] = dfGender['Age'].astype(str)
data = defaultdict(list)


data = np.array(dfGender['Age']).tolist()

#print(data)


te = TransactionEncoder()
te_data = te.fit_transform(data)
data = pd.DataFrame(te_data, columns=te.columns_)

frequent_itemsets = apriori(data, min_support=0.5, use_colnames=True)
print(frequent_itemsets)







     support      itemsets
0   0.961151           (-)
1   0.581000           (2)
2   0.599913           (3)
3   0.973322           (5)
4   0.683719           (6)
5   0.581000        (-, 2)
6   0.599913        (-, 3)
7   0.934473        (-, 5)
8   0.683719        (-, 6)
9   0.581000        (5, 2)
10  0.599913        (5, 3)
11  0.599913        (6, 3)
12  0.683719        (5, 6)
13  0.581000     (-, 5, 2)
14  0.599913     (-, 5, 3)
15  0.599913     (-, 6, 3)
16  0.683719     (5, -, 6)
17  0.599913     (5, 6, 3)
18  0.599913  (5, -, 6, 3)


La tabla muestra el soporte y los itemsets (conjuntos de elementos) que aparecen en cada regla. Por ejemplo, la regla número 2 indica que el itemset (conjunto de elementos) 3 aparece con un soporte de 0.599913, lo que significa que el 59.99% de las transacciones incluyen el elemento 3.

También se pueden ver reglas que combinan varios elementos, como la regla número 16 que indica que el itemset (conjunto de elementos) formado por los elementos 5 y 6 aparece junto con el elemento "-" (que puede representar cualquier otro elemento) con un soporte del 68.37%.

En resumen, estas tablas muestran las relaciones y patrones encontrados en el conjunto de datos, que pueden ser útiles para entender la estructura y relaciones entre los elementos en el dataset.